rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+React.HTMLAttributes<HTMLElement>
+```
+
+## Property overrides of native properties
+
+> Several components declare properties that are also declared by extending from the DOM types (e.g. React.HtmlAttributes<T>).
+
+checked, children, color, defaultChecked, disabled, href, image, name, onClick, open, ref, target, type, value
+
+> Several components use declar properties that are also HTML element names.
+
+button, label strong
+
+# Layout properties
+
+> Positioning is defined differently based on component layout specifics.
+> Some props define a type for their enums like start | end, other's don't.
+> Some props leverage Popper positioning.
+
+AccordionHeaderCommons
+
+```ts
+// AccordionHeaderExpandIconPosition = 'start' | 'end'
+expandIconPosition: AccordionHeaderExpandIconPosition;
+inline: boolean;
+```
+
+BadgeProps
+
+```ts
+iconPosition: 'before' | 'after';
+```
+
+ButtonProps
+
+```ts
+iconPosition: 'before' | 'after';
+```
+
+DividerProps
+
+```ts
+alignContent: 'start' | 'end' | 'center';
+```
+
+ImageProps
+
+```ts
+fit: 'none' | 'center' | 'contain' | 'cover';
+```
+
+LinkProps
+
+```ts
+inline: boolean;
+```
+
+MenuProps, PopoverProps, TooltipProps
+
+```ts
+// PositioningShorthand = PositioningProps | PositioningShorthandValue
+// PositioningShorthandValue = 'above'  | 'above-start'  | 'above-end'  | below'  | 'below-start'  | 'below-end'  | 'before'  | 'before-top'  | before-bottom'  | 'after'  | 'after-top'  | 'after-bottom'
+positioning: PositioningShorthand;
+```
+
+## Size properties
+
+> Some components use different sets of semantic sizes. Avatar uses a set of numbers.
+> The property name for filling available space is sometimes block and sometimes fluid. The fit property value of cover could also be considered filling available space.
+
+AccordionHeaderCommons
+
+```ts
+// AccordianHeaderSize = 'small' | 'medium' | 'large' | 'extra-large'
+size?: AccordionHeaderSize;
+```
+
+AvatarProps
+
+```ts
+// AvatarSizeValue = 20 | 24 | 28 | 32 | 36 | 40 | 48 | 56 | 64 | 72 | 96 | 120 | 128
+size: AvatarSizeValue;
+```
+
+BadgeProps
+
+```ts
+// BadgeSize = 'smallest' | 'smaller' | 'small' | 'medium' | 'large' | 'larger' | 'largest'
+
+size: BadgeSize;
+showZero: boolean;
+```
+
+CounterBadgeProps
+
+```ts
+showZero: boolean;
+```
+
+ButtonProps
+
+```ts
+block: boolean;
+size: 'small' | 'medium' | 'large';
+```
+
+ImageProps
+
+```ts
+fit: 'none' | 'center' | 'contain' | 'cover';
+fluid: boolean;
+```
+
+LabelProps
+
+```ts
+size: 'small' | 'medium' | 'large';
+```
+
+PopoverProps
+
+```ts
+//PopoverSize = 'small' | 'medium' | 'large'
+size: PopoverSize;
+```
+
+## Appearance properties
+
+> Some components have mutually exclusive booleans and others use discriminated unions.
+> Some components define a color property that are a set of colors, others are a set semantic names.
+> Some naming is inconsistent (e.g. noArrow, pointing)
+
+AvatarProps
+
+```ts
+activeDisplay: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
```

Comment:
I'm wondering if there should be two prop here:  One for that lets you specify 'ring' and maybe future values and another for 'shadow' | 'glow'.  Are these specific appearances or is this union trying to cover the permutations of two different things?

Reply:
Per chat with Levi, we should break these up if they are enumerating permutations.  I'll follow up with Avatar owner to find out.